librz/bin/p/bin_luac.c

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// SPDX-FileCopyrightText: 2021 Heersin <teablearcher@gmail.com>
+
+#include <rz_bin.h>
+#include <rz_lib.h>
+#include "librz/bin/format/luac/luac_common.h"
+
+#define GET_INTERNAL_BIN_INFO_OBJ(bf) ((LuacBinInfo *)(bf)->o->bin_obj)
+
+static bool check_buffer(RzBuffer *buff) {
+	if (rz_buf_size(buff) > 4) {
+		ut8 buf[LUAC_MAGIC_SIZE];
+		rz_buf_read_at(buff, LUAC_MAGIC_OFFSET, buf, LUAC_MAGIC_SIZE);
+		return !memcmp(buf, LUAC_MAGIC, LUAC_MAGIC_SIZE);
+	}
+	return false;
+}
+
+static bool load_buffer(RzBinFile *bf, void **bin_obj, RzBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	ut8 MAJOR_MINOR_VERSION;
+	LuacBinInfo *bin_info_obj = NULL;
+	LuaProto *proto = NULL;
+	RzBinInfo *general_info = NULL;
+	st32 major;
+	st32 minor;
+
+	rz_buf_read_at(buf, LUAC_VERSION_OFFSET, &MAJOR_MINOR_VERSION, sizeof(MAJOR_MINOR_VERSION)); /* 1-byte in fact */
+	if ((bin_info_obj = RZ_NEW(LuacBinInfo)) == NULL) {
+		return false;
+	}
+	major = (MAJOR_MINOR_VERSION & 0xF0) >> 4;
+	minor = (MAJOR_MINOR_VERSION & 0x0F);
+
+	if (major != 5) {
+		eprintf("currently support lua 5.x only\n");
+		return NULL;
+	}
+
+	// TODO : find a way to get whole buffer
```

Comment:
The whole purpose of `RzBuffer` is to try to avoid having to load into memory the entire file, because if that's huge, it is going to be a problem. Is it possible to get basic lua information that you need here without reading the entire input file but just pieces of it.

`rz_buf_read`(and its variants) should be preferred most of the time.
If you *really* (and I mean *really*) need to get the whole data, you can use `rz_buf_data`

Reply:
Thanks, I understand the purpose of RzBuffer. I reviewed my code and I think it's possible to refactor it with rz_buf_read.

Reply:
Now the read operations are refactored to rz_buf_read